### ISSUE TITLE
Pinch-zoom with tap functionality on mobile

### ISSUE THREAD

**Issue by BG687** [konvajs/konva#577]
Hi, I can state this bug clearly without the need for a demo. I'm sure others have encountered this before. 

I am building a mobile responsive canvas with an image. Users need to be able to pinch-zoom into the image and place pins (which are placed through a tap and hold gesture). I have enabled default behavior to allow for this. The problem is that I cannot distinguish between a pinch-zoom (two fingers tapping at once) and a tap and hold. 

Part of the problem has to do with the touchstart and touchend events. In theory, I could use these to determine if taps are occurring simultaneously. However, these event callbacks do not fire reliably - sometimes on a pinch-zoom, the touchend event will only fire once instead of twice, for example. This is unexpected, because a pinch-zoom requires two fingers to be touching the screen at different points on the canvas.  

So, basically, I need to be able to reliably distinguish between  a pinch-zoom and a single tap. And ideas here are welcome. Also, it would probably be good to mention the unreliability of the touchstart / touchend callbacks in the documentation. 

Thanks

**Comment by lavrton** [comment]
I think the root of the issue is that current `Konva` can't handle multitouch correctly. I hope to rewrite the event system on `Konva` so we can have the full support of touch gestures.

**Comment by trip1** [comment]
This would be incredibly useful.

**Comment by lavrton** [comment]
There was a large update to support multitouch events for Konva.
I guess this issue should be resolved now. If not - please add more info.
